devops/setup-gpg/action.yaml

```diff
@@ -0,0 +1,48 @@
+name: 'Setup GPG'
+description: 'Configures this action to run gpg with a given key and pass'
+inputs:
+  private-key:  # id of input
+    description: 'gpg private key'
+    required: true
+  key-pass:  # id of input
+    description: 'gpg key pass'
+    required: true
+  key-name:  # id of input
+    description: 'gpg key name'
+    required: true
+    default: "Aerospike"
+runs:
+  using: "composite"
+  steps:
+      - name: Set up GPG
+        env:
+          GPG_PRIVATE_KEY: ${{ inputs.private-key }}
+          GPG_KEY_PASS: ${{ inputs.key-pass }}
+          GPG_ID: ${{ inputs.key-name }}
+        run: |
+          # Setup gpg
```

Comment:
I would define separate steps for each of the 4 sections you have here:
- Setup gpg
- Configure gpg for non-interactive use
etc.

It makes it easier to tell which exact part of the code might fail